### PR TITLE
fix autoscaling due to VMSS tag prefix issue

### PR DESCRIPTION
**Pull request by m-craghead**
#### Which component this PR applies to?
cluster-autoscaler

#### What type of PR is this?
/kind bug

/kind regression

#### What this PR does / why we need it:
Fixes a bug 
#### Which issue(s) this PR fixes:

Fixes # Azure AKS cluster-autoscaler non-functional duo to recently added tag-prefix #4666

#### Does this PR introduce a user-facing change?

```release-note NONE

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: cluster-autoscaler/cloudprovider/azure/azure_kubernetes_service_pool.go [hunk 1]
```diff
@@ -315,7 +315,7 @@ func (agentPool *AKSAgentPool) DeleteNodes(nodes []*apiv1.Node) error {
 
 //IsAKSNode checks if the tag from the vm matches the agentPool name
 func (agentPool *AKSAgentPool) IsAKSNode(tags map[string]*string) bool {
-	poolName := tags["poolName"]
+	poolName := tags["aks-managed-poolname"]
```
Comment (marwanad) [inline cluster-autoscaler/cloudprovider/azure/azure_kubernetes_service_pool.go:318]: can we move this to a const at the start of the file? The tag should be `aks-managed-poolName`

**Comment by marwanad** [comment]
You need to fix the test case under: azure_kubernetes_sercice_pool_test as well and while you're at it - can we fix the filename? "sercice"

**Comment by marwanad** [comment]
So, the label only applies to new AKS clusters (New nodepools to be strictly correct)

The old clusters still have `poolName` so we need to come up with some regex or a smarter way to accommodate both.

File: cluster-autoscaler/cloudprovider/azure/azure_kubernetes_service_pool.go [hunk 2]
```diff
@@ -315,7 +319,10 @@ func (agentPool *AKSAgentPool) DeleteNodes(nodes []*apiv1.Node) error {
 
 //IsAKSNode checks if the tag from the vm matches the agentPool name
 func (agentPool *AKSAgentPool) IsAKSNode(tags map[string]*string) bool {
-	poolName := tags["poolName"]
+	poolName := tags[aksManagedPoolName]
+	if poolName == nil {
+		poolName = tags["poolName"]
```
Comment (marwanad) [inline cluster-autoscaler/cloudprovider/azure/azure_kubernetes_service_pool.go:324]: keep it as a constant:

`legacyAKSPoolNameTag`

and `aksManagedPoolNameTag`

**Comment by nilo19** [comment]
can you squash the commits?

**Comment by marwanad** [comment]
@m-craghead can you create cherry-picks for 1.23,1.22 and 1.21?

**Pull request merged**